### ISSUE TITLE
pyright rejects declaration of generic protocol but still understands it

### ISSUE THREAD

**Issue by untitaker** [microsoft/pyright#2276]
**Describe the bug**
A declaration of a generic (callback) protocol will lead pyright to emit an error for using a non-protocol baseclass `Generic`. However, besides that error pyright behaves correctly in usage of that protocol.

**To Reproduce**

In pyright 1.1.166 running as part of coc.nvim, open this file and observe:

```
from typing import Protocol, Generic, TypeVar

ConcreteInt = TypeVar("ConcreteInt", bound=int, contravariant=True)

# pyright errors: 
# [Pyright] [E] Protocol class "Type[IntFunction[ConcreteInt@IntFunction]]" cannot derive from non-protocol class "Type[Generic[ConcreteInt@IntFunction]]"
class IntFunction(Protocol, Generic[ConcreteInt]):
    def __call__(self, myint: ConcreteInt) -> None:
        ...


class SubInt(int): pass


def bar(myint: SubInt) -> None:
    pass

# fails in mypy and pyright as expected because SubInt != int
foo: IntFunction[int] = bar

# passes in mypy and pyright as expected
foo2: IntFunction[SubInt] = bar
```

**Expected behavior**

pyright should not error near the declaration of IntFunction but only near `foo`.

**Screenshots or Code**

![Screenshot 2021-09-07 at 19 29 46](https://user-images.githubusercontent.com/837573/132386715-920625c6-35c3-4073-b193-39e7ceab1dca.png)

**VS Code extension or command-line**

I'm using pyright as part of coc.nvim, pyright version is 1.1.166

**Comment by erictraut** [comment]
It appears that `Protocol` special-cases `Generic` when it performs its multi-inheritance check, so I agree this is a false positive. I don't think it has come up before because generic type variables are typically specified as subscripts to `Protocol`, like this:

```python
class IntFunction(Protocol[ConcreteInt]):
    ...
```

**Comment by erictraut** [comment]
This will be fixed in the next release.

**Comment by untitaker** [comment]
Thanks for the workaround with `Protocol[]`, always learning things

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.167, which I just published. It will also be included in the next release of pylance.

**Comment by untitaker** [comment]
Thank you for your service 🙇
